src/terrain/generation/land_generator.cpp

```diff
@@ -51,127 +58,33 @@ LandGenerator::get_num_stamps(const Json::Value& biome) {
 
 TileStamp
 LandGenerator::get_this_stamp() const {
-    TileStamp out;
-    out.mat = &materials.at(data_[current_region]["Material_id"].as<int>());
-    out.color_id = data_[current_region]["Color_id"].asInt();
-    for (Json::Value::ArrayIndex i = 0; i < data_[current_region]["Can_Stamp"].size();
-         i++) {
-        int E = data_[current_region]["Can_Stamp"][i]["E"].asInt();
-        if (data_[current_region]["Can_Stamp"][i]["C"].isInt()) {
-            int C = data_[current_region]["Can_Stamp"][i]["C"].asInt();
-            out.elements_can_stamp.insert(std::make_pair(E, C));
-        } else if (data_[current_region]["Can_Stamp"][i]["C"].asBool()) {
-            for (unsigned int C = 0; C < materials.at(E).color.size(); C++) {
-                out.elements_can_stamp.insert(std::make_pair(E, C));
-            }
-        }
-    }
-    std::string type = data_[current_region]["Type"].asString();
-
-    if (type == "Positions") {
-        from_positions(current_region, current_sub_region, out);
-    } else if (type == "Radius") {
-        from_radius(current_region, current_sub_region, out);
-    } else if (type == "Grid") {
-        from_grid(current_region, current_sub_region, out);
-    }
+    TileStamp out = stamp_generators_[current_region]->operator()(current_sub_region);
```

Comment:
```suggestion
    TileStamp out = stamp_generators_[current_region]->stamp_region(current_sub_region);
```

No point in using `operator()` if you call it like this.